### ISSUE TITLE
How much time is needed to train the model?

### ISSUE THREAD

**Issue by ostreech1997** [deepmipt/DeepPavlov#1210]
Hi!
I use your library to build chatbot. I use RuBERT model for NER task, and rusentiment BERT for classification task. I want to create progress bar to track train process. So, is it possible to find out the time needed to train the model? Or may be, there is another way to get information of train process progress.

**Comment by dilyararimovna** [comment]
Hello!
You can always change logging parameters in config file (example from [here).](https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/configs/classifiers/rusentiment_convers_bert.json)
```
"train": {
    "batch_size": 64,
    "epochs": 100,
    "metrics": [
      "f1_weighted",
      "f1_macro",
      "accuracy",
      {
        "name": "roc_auc",
        "inputs": [
          "y_onehot",
          "y_pred_probas"
        ]
      }
    ],
    "show_examples": false,
    "pytest_max_batches": 2,
    "validation_patience": 5,
    "val_every_n_epochs": 1,
    "log_every_n_epochs": 1,
    "evaluation_targets": [
      "train",
      "valid",
      "test"
    ],
    "tensorboard_log_dir": "{MODEL_PATH}/"
  },
```
First of all, you can change `validation_patience` (that equals to number of validation procedures with not-improved validation target metrics before stop training), parameters `val_every_n_epochs` and `log_every_n_epochs` could be also changed and by the way you can use `val_every_n_batches` and `log_every_n_batches` to validate and log more frequently.
Finally, you may use tensorboard to look into plots of metrics.

**Comment by ostreech1997** [comment]
Hi @dilyararimovna, thanks for your responce!
Sorry, I don't understan how to use your advice.... You offer me to use logs somehow to track learning progress? Do I have to track number of epochs done? I thought about it, but there are situations, when model is stopped earlier than 100 epochs, how I can overcome this problem?

**Comment by dilyararimovna** [comment]
If you have a validation set, you may track the accuracy (this is an example, you may track any metric). So, you just need to decide how often to check it. For example, every 1 epoch:

"train": {
    "batch_size": 64,
    "epochs": 100,
    "metrics": [
      "accuracy"
    ],
    "show_examples": false,
    "pytest_max_batches": 2,
    "validation_patience": 5,
    "val_every_n_epochs": 1,
    "log_every_n_epochs": 1,
    "evaluation_targets": [
      "train",
      "valid",
      "test"
    ],
    "tensorboard_log_dir": "{MODEL_PATH}/"
  },

**Comment by ostreech1997** [comment]
Возможно мой английский очень плох....))
Меня интересует не отслеживание качества модели. Я хочу сделать прогресс бар времени обучения. То есть, чтобы я запускал обучение и видел в реальном времени сколько еще осталось учиться модели. Как вариант, я думал считать процент обучения с помощью пройденных эпох (epochs_done/epochs*100), но модель никогда не проходит все эпохи. Может быть в вашем фрейморке предусмотрено что-то, что поможет мне понять, сколько времени еще осталось обучаться модели?

**Comment by dilyararimovna** [comment]
Нельзя точно гарантировать, сколько модели осталось обучаться в случае, если вы используете early stopping по метрикам на валидации. Поэтому смысла от progress bar в таком случае нет.

Однако вы можете попробовать использовать tensorboard или смотреть в сами логи (первое нагляднее), чтобы смотреть как ведут себя метрики, видна ли уже сходимость на трейне и как ведут себя метрики на валидации, что будет в некоторым смысле хорошей оценкой, сколько еще осталось тренироваться вашей модели. (в tensorboard есть графики по времени, то есть можно наглядно оценивать)

**Comment by ostreech1997** [comment]
Понятно
@dilyararimovna спасибо большое за ответ!
